### PR TITLE
JAMES-4044 LDAP attribute matchers

### PR DESCRIPTION
**Pull request by chibenwa**

### CODE REVIEW

File: server/mailet/ldap/README.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+ - LDAP conditions based on attributes of the sender
+
+Support is planned for LDAP groups both for sender and recipients.
+
+## Set up
+
+Build this project. For the root of the repository:
+
+```bash
+mvn clean isntall -DskipTests --am --pl :james-server-mailet-ldap
```
Comment (quantranhong1999) [inline server/mailet/ldap/README.md:18]:
```suggestion
mvn clean install -DskipTests --am --pl :james-server-mailet-ldap
```

File: server/apps/distributed-app/docs/modules/ROOT/partials/LDAPMatchers.adoc [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+ - LDAP conditions based on attributes of the sender
+
+Support is planned for LDAP groups both for sender and recipients.
+
+== Set up
+
+Build this project. For the root of the repository:
+
+....bash
+mvn clean isntall -DskipTests --am --pl :james-server-mailet-ldap
```
Comment (quantranhong1999) [inline server/apps/distributed-app/docs/modules/ROOT/partials/LDAPMatchers.adoc:18]:
```suggestion
mvn clean install -DskipTests --am --pl :james-server-mailet-ldap
```

File: server/mailet/ldap/src/main/java/org/apache/james/transport/matchers/HasLDAPAttribute.java [hunk 3]
```diff
@@ -44,6 +44,30 @@
 import com.unboundid.ldap.sdk.SearchResultEntry;
 import com.unboundid.ldap.sdk.SearchScope;
 
+/**
+ * Matchers that allow looking up for LDAP attributed for each recipient.
+ *
+ * To enable this matcher one need first to add the james-server-mailet-ldap.jar in the externals-jars folder of your
+ * James installation.
+ *
+ * In order to match the presence of an attibute:
```
Comment (quantranhong1999) [inline server/mailet/ldap/src/main/java/org/apache/james/transport/matchers/HasLDAPAttribute.java:53]:
```suggestion
 * In order to match the presence of an attribute:
```

**Review by quantranhong1999 — APPROVED** [review]

File: server/mailet/ldap/README.md [hunk 4]
```diff
@@ -18,4 +18,24 @@ Build this project. For the root of the repository:
...
+Then copy the JAR located in `/target` into the `extensions-jars` folder of your Apache James installation.
+
+## Configuration
+
+The LDAP settings are reused from the `usersrepository.xml` configuration file.
+
+Then LDAP matchers can be configured within `mailetcontainer.xml`. For instance:
+
+```xml
+<mailet matcher="HasLDAPAttibute=description:blocked" class="Null">
```
Comment (Arsnael) [inline server/mailet/ldap/README.md:33]:
```suggestion
<mailet matcher="HasLDAPAttribute=description:blocked" class="Null">
```

**Review by Arsnael — COMMENTED** [review]
Really can't let the typo in the doc on the matcher name itself in examples pass sorry :)

**Review by Arsnael — APPROVED** [review]

**Comment by Arsnael** [comment]
Little squash needed :)

**Review by quantranhong1999 — APPROVED** [review]

**Comment by Arsnael** [comment]
Still seeing bunch of fixup commits, didn't get squashed

**Pull request merged**